microprofile/reactive-messaging-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/reactive/messaging/deployment/ReactiveMessagingDependencyProcessor.java

```diff
@@ -184,4 +192,13 @@ private boolean allowExperimentalAnnotations() throws DeploymentUnitProcessingEx
         }
         return experimental;
     }
+
+    @SuppressWarnings("deprecation")
+    private void ignorePrecalulatedJandex(DeploymentUnit deploymentUnit, String... modules) {
+        final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
+        if (support.hasCapability(WELD_CAPABILITY_NAME)) {
+            WeldCapability weld = support.getOptionalCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class).get();
```

Comment:
That's a clever way to avoid dealing with the NoSuchCapabilityException!

Reply:
Copied and pasted from JSF :-D